### PR TITLE
Introduce setup.cfg and black+isort

### PR DESCRIPTION
**Pull request by eggplants**
Close: #257

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+extend-ignore =
+    # E203 whitespace before ':'
+    # https://github.com/psf/black/issues/315
+    E203
+per-file-ignores =
+    tests/test_stubs.py:F821
+    tests/test_typing.py:F821
+
+[mypy]
+python_version = 3.6
```
Comment (carljm) [inline setup.cfg:72]: this should also be 3.7

**Review by carljm — APPROVED** [review]
Looks great! Thank you for doing this! Just should bump the min Python version up to 3.7 a couple places, but I'd also be fine if that happens in a separate PR.

File: setup.cfg (hunk 1 above)
Comment (eggplants) [reply setup.cfg]: da80482

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    Programming Language :: Python :: 3
+    Programming Language :: Python :: 3.6
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
+    Programming Language :: Python :: 3.9
+
+[options]
+packages = find:
+include-package-data = True
+python_requires = >= 3.6
```
[the remark this answers is not in the document]
Comment (eggplants) [reply setup.cfg]: da80482

**Pull request merged**